src/clients/HttpClient.ts

```diff
@@ -13,7 +13,7 @@ class HttpClient implements IHttpClient {
         fullRoute?: boolean
     ): Promise<Response> {
         if (!fullRoute && this.serverUrl) {
-            route = this.basePath + route;
+            route = this.serverUrl + this.basePath + route;
```

Comment:
BTW, if you're including the locale JSONs in the build (I)HttpClient shouldn't be necessary anymore. Just rework ServerClient to take the server URL directly instead.